testcases/PowerNVDump.py

```diff
@@ -561,12 +618,127 @@ def runTest(self):
+            self.c.run_command("sed -i 's/-l --message-level/-l -F --message-level/' /etc/kdump.conf;")
+            self.c.run_command("sed -i '/path \/var\/crash/c\path %s' /etc/kdump.conf;" % self.net_path)
+            self.c.run_command("cd /root; kdumpctl propagate", timeout=600)
+            self.c.run_command("kdumpctl restart", timeout=600)
+            time.sleep(5)
+            res = self.c.run_command("service kdump status | grep active")
+            if 'exited' not in res[0].strip():
+                print("Kdump service is not configured properly")
+        elif self.distro == "ubuntu":
+            self.c.run_command("sed -i '/SSH=\"<user at server>\"/c\SSH=\"root@%s\"' /etc/default/kdump-tools" % self.server_ip)
+            self.c.run_command("sed -i '/SSH_KEY=\"<path>\"/c\SSH_KEY=/root/.ssh/id_rsa' /etc/default/kdump-tools")
+            self.c.run_command("kdump-config unload;")
+            self.c.run_command("kdump-config load;")
+            time.sleep(5)
+        else:
+            self.c.run_command('sed -i \'/^KDUMP_SAVEDIR=/c\KDUMP_SAVEDIR=\"ssh:\/\/root:%s@%s\/%s\"\' /etc/sysconfig/kdump;' % (self.server_pw, self.server_ip, self.net_path))
+            self.c.run_command("touch /etc/sysconfig/kdump; systemctl restart kdump.service; sync", timeout=600)
+            time.sleep(5)
+
+
+    def setup_nfs(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        if self.distro == "rhel":
+            self.c.run_command("sed -e '/ssh/ s/^#*/#/' -i /etc/kdump.conf;")
+            self.c.run_command("yum -y install nfs-utils")
+            self.c.run_command("service nfs-server start")
+            self.c.run_command("echo 'nfs %s:%s' >> /etc/kdump.conf;" % (self.server_ip, self.net_path))
+            self.c.run_command("sed -i 's/-l --message-level/-l -F --message-level/' /etc/kdump.conf;")
+            self.c.run_command("sed -i '/path \/var\/crash/c\path \/' /etc/kdump.conf;")
+            self.c.run_command("mount -t nfs %s:%s /var/crash" % (self.server_ip, self.net_path))
+            self.c.run_command("cd /root; kdumpctl restart", timeout=600)
+            res = self.c.run_command("service kdump status | grep active")
+            if 'exited' not in res[0].strip():
+                print("Kdump service is not configured properly")
+        elif self.distro == "ubuntu":
+            self.c.run_command("apt-get install -y nfs-common;")
+            self.c.run_command("apt-get install -y nfs-kernel-server;")
+            self.c.run_command("service nfs-server start;")
+            self.c.run_command("mount -t nfs %s:%s /var/crash;" % (self.server_ip, self.net_path))
+            self.c.run_command("sed -e '/SSH/ s/^#*/#/' -i /etc/default/kdump-tools;")
+            self.c.run_command("sed -i '/NFS=\"<nfs mount>\"/c\\NFS=\"%s:%s\"' /etc/default/kdump-tools" % (self.server_ip, self.net_path))
+            time.sleep(5)
+        else:
```

Comment:
Since it was checked initially, thought if i can skip that check.

        res = self.c.run_command("cat /etc/os-release")
        if "Ubuntu" in res[0] or "Ubuntu" in res[1]:
            self.distro = "ubuntu"
        elif 'Red Hat' in res[0] or 'Red Hat' in res[1]:
            self.distro = 'rhel'
        elif 'SLES' in res[0] or 'SLES' in res[1]:
            self.distro = 'sles'
        else:
            raise self.skipTest("Test currently supported only on ubuntu, sles and rhel")